src/Symfony/Component/Scheduler/RecurringMessage.php

```diff
@@ -51,6 +53,21 @@ public static function trigger(TriggerInterface $trigger, object $message): self
         return new self($trigger, $message);
     }
 
+    /**
+     * Unique identifier for this message's context.
+     */
+    public function getId(): string
+    {
+        if (isset($this->id)) {
+            return $this->id;
+        }
+
+        return $this->id = sha1(implode('', [
+            $this->message instanceof \Stringable ? (string) $this->message : $this->message::class,
+            $this->trigger instanceof \Stringable ? (string) $this->trigger : $this->trigger::class,
```

Comment:
I would always use the class name as the string representation can be the same for 2 triggers or messages.